packages/edit-site/src/components/editor/index.js

```diff
@@ -152,6 +157,11 @@ function Editor() {
 		onClose: () => setIsInserterOpened( false ),
 	} );
 
+	// Don't render the Editor until the settings are set and loaded
+	if ( ! settings?.siteUrl ) {
```

Comment:
I'm just curious about this. Was it needed because something was breaking or an optimization?

Reply:
Yes, rendering the `Editor` component without settings results in crash.